src/main/java/server/router/RequestRouter.java

```diff
@@ -31,14 +31,15 @@ public Response getResponse(Request request) {
         return route.processRequest();
       }
     }
+
     return null;
   }
 
   private List<Route> getAllRoutes() {
     return List.of(
+            new HeadRequest(request),
```

Comment:
Each time you need to find a route, you will need to loop over this list of created instances. Is there another way to get the correct handler without creating an instance first? Maybe using other data structure that provide you more efficient O(1) access time?

Reply:
Ah I think a HashMap has a O(1) complexity so I'll try that!

Reply:
Looks much better now!